chef/cookbooks/provisioner/templates/default/crowbar_join.suse.sh.erb

```diff
@@ -63,6 +63,28 @@ log_to() {
     return $_ret
 }
 
+get_state() {
+  local output
+  local curlargs=(--connect-timeout 60 -s -L \
+      -H "Accept: application/json" -H "Content-Type: application/json" \
+      --max-time 240 --insecure --location)
+  [[ $CROWBAR_KEY ]] && curlargs+=(-u "$CROWBAR_KEY" --digest --anyauth)
+  oldumask=`umask`
+  umask 077
```

Comment:
so why do we set this? Its the first time I find this in a script. Is it necessary for the curl to work?

Reply:
TBH, I just copied what is done for `post_state` :-)

Reply:
It only makes sense when you create (e.g. download) files, so that they will not be world-readable upon creation.
But the curl below only produces stdout, so the umask changes can be dropped here.

Reply:
There you go.